tests/nn/util_test.py

```diff
@@ -437,6 +437,30 @@ def test_viterbi_decode(self):
         assert indices == [3, 2, 1]
         assert value.numpy() == 18
 
+        # Test that providing evidence results in paths containing specified tags.
+        sequence_predictions = torch.FloatTensor([[0, 0, 0, 4, 4],
```

Comment:
Seeing 4s here and the "4th" tag, along with the name "predictions", got me a little confused.  I think it'd be a bit more readable if this were named `sequence_logits`, and if you used a number that is not 4 or 5 for the logit values, so there's less room for garden path readings of the comment below.